File: mobile/listener_app/lib/models/session.dart
import 'enums.dart';

class Session {
  Session({
    required this.id,
    required this.userId,
    required this.eventId,
    required this.channelId,
    required this.role,
    required this.status,
    required this.startedAt,
    this.endedAt,
  });

  final String id;
  final String userId;
  final String eventId;
  final String channelId;
  final SessionRole role;
  final SessionStatus status;
  final DateTime startedAt;
  final DateTime? endedAt;

  factory Session.fromJson(Map<String, dynamic> json) {
    return Session(
      id: json['id'] as String,
      userId: json['userId'] as String,
      eventId: json['eventId'] as String,
      channelId: json['channelId'] as String,
      role: SessionRoleApi.fromApiValue(json['role'] as String? ?? 'Listener'),
      status: SessionStatusApi.fromApiValue(json['status'] as String? ?? 'Active'),
      startedAt: DateTime.parse(json['startedAt'] as String),
      endedAt: json['endedAt'] != null ? DateTime.parse(json['endedAt'] as String) : null,
    );
  }
}

File: backend/Backend.Models/Entities/Session.cs
using Backend.Models.Enums;

namespace Backend.Models.Entities;

public sealed class Session
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid EventId { get; set; }
    public Guid ChannelId { get; set; }
    public SessionRole Role { get; set; }
    public DateTime StartedAt { get; set; }
    public DateTime? EndedAt { get; set; }
    public SessionStatus Status { get; set; }

    public User User { get; set; } = null!;
    public Event Event { get; set; } = null!;
    public Channel Channel { get; set; } = null!;
    public ICollection<Producer> Producers { get; set; } = new List<Producer>();
    public ICollection<Consumer> Consumers { get; set; } = new List<Consumer>();
    public ICollection<Recording> Recordings { get; set; } = new List<Recording>();
    public ICollection<Caption> Captions { get; set; } = new List<Caption>();
    public ICollection<Transport> Transports { get; set; } = new List<Transport>();
}
